functions/go/export-terraform/README.md

```diff
@@ -0,0 +1,90 @@
+# export-terraform-resources
+
+## Overview
+
+<!--mdtogo:Short-->
+
+Generate idiomatic Terraform configuration equivalents for Config Connector resources.
+
+<!--mdtogo-->
+
+This function generates idiomatic Terraform configuration by parsing [Config Connector (KCC)](https://cloud.google.com/config-connector/docs) resources and pushing the equivalent Terraform configuration into a `ConfigMap`.
+
+Where appropriate, the generated Terraform references [Cloud Foundation Toolkit modules](https://g.co/dev/terraformfoundation).
+The goal is to make the generated output as close to possible as what a human would have written.
+
+The output Terraform will be saved to a `ConfigMap` in `terraform.yaml` at the root of the package.
+Each key in the `ConfigMap` corresponds to a different file which is part of the Terraform module.
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: terraform
+  annotations:
+    config.kubernetes.io/local-config: "true"
+    blueprints.cloud.google.com/syntax: "hcl"
+    blueprints.cloud.google.com/flavor: "terraform"
+data:
+  folders.tf: |+
+    resource "google_folder" "test" {
+      display_name = "Test Display"
+      parent       = "organizations/123456789012"
+    }
+```
+
+<!--mdtogo:Long-->
+
+## Usage
+
+The function executes as follows:
+
+1. Searches for supported KCC resources in the package
+2. For all KCC resources found, generate the equivalent Terraform
+3. Output the Terraform into a `ConfigMap`.
+
+`export-terraform` function can be executed imperatively as follows:
```

Comment:
If the imperative workflow is the only way we want to support, why do we choose to output in a ConfigMap instead of mounting a volume and output the file there?
If desired, we can support both.

Reply:
A ConfigMap is both more portable (can be served over HTTP APIs easily) and easier to generate (we don't need to give the function filesystem access).

The eventual output which we want for customers is a zip file, but I don't want to entangle zip file generation into this function. See go/terraform-krm-export for more background.